test/functional/apps/visualize/groups.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+export default function (): any {
+  const xs = new Map();
```

Comment:
What's the logic of adding this?  Is it going to make it easier to maintain or prevent duplicates somehow?

Reply:
Well if we really care about preventing dupes, this can be a `Set` instead of a `Map`.

Reply:
It's also about making the code more `DRY`.  Removing duplicated calls, that can use list processing instead.